shared/linkify.ts

```diff
@@ -73,12 +49,25 @@ linkify.add("web+", {
 	},
 	normalize(match) {
 		match.schema = match.text.slice(0, match.text.indexOf(":") + 1);
-		LinkifyIt.prototype.normalize(match); // hand over to the global override
+	},
+});
+
+// we must rewrite protocol less urls to http, else if TL is hosted
+// on https, this would incorrectly use https for the remote link.
+// See https://github.com/thelounge/thelounge/issues/2525
+//
+// We take the validation logic from linkify and just add our own
+// normalizer.
+linkify.add("//", {
+	validate: (linkify as any).__schemas__["//"].validate,
```

Comment:
nasty? Yes. This is using a private implementation detail.
But our tests will catch it, should this every blow up.